api/v1alpha1/cephcluster_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CephConfig map[string]string
+
+// ReadAffinitySpec capture Ceph CSI read affinity settings
+type ReadAffinitySpec struct {
+	CrushLocationLabels []string `json:"crushLocationLabels,omitempty"`
+}
+
+// CephClusterSpec defines the desired state of CephCluster
+type CephClusterSpec struct {
+	Monitors             []string              `json:"monitors"`
+	ReadAffinity         ReadAffinitySpec      `json:"readAffinity,omitempty"`
+	RbdMirrorDaemonCount int                   `json:"rbdMirrorDaemonCount,omitempty"`
+	CephConfig           map[string]CephConfig `json:"cephConfig,omitempty"`
```

Comment:
Let's move this to a configmap where the user will get the configmap name in the CR. WDYT?